packages/utils/src/Cache.ts

```diff
@@ -0,0 +1,21 @@
+export class Cache<V> {
+
+    private value?: V
+    private valueTimestamp?: number
+    private readonly valueFactory: () => Promise<V>
+    private readonly maxAgeInMilliseconds: number
+
+    constructor(valueFactory: () => Promise<V>, maxAgeInMilliseconds: number) {
+        this.valueFactory = valueFactory
+        this.maxAgeInMilliseconds = maxAgeInMilliseconds
+    }
+
+    async get(): Promise<V> {
+        const now = Date.now()
+        if ((this.valueTimestamp === undefined) || (now > (this.valueTimestamp + this.maxAgeInMilliseconds))) {
+            this.valueTimestamp = now
+            this.value = await this.valueFactory()
```

Comment:
Would it be acceptable for `valueFactory` to reject? If so I would swap the order of the statements here
```suggestion
            this.value = await this.valueFactory()
            this.valueTimestamp = now
```